src/RecursiveApply/RecursiveApply.jl

```diff
@@ -9,17 +9,60 @@ module RecursiveApply
 
 export ⊞, ⊠, ⊟
 
+# These functions need to be generated for type stability (since T.parameters is
```

Comment:
Let's add these changes (in RecursiveApply) as a separate PR, with tests.

Reply:
Should I include `rmul_with_projection` in that PR? It would probably make more sense to have it be part of RecursiveApply, though that would require loading the RecursiveApply module after the Geometry module.

Reply:
No, just the RecursiveApply part. Projections aren't loaded until Geometry, so better keep that where it is for now.